airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -232,6 +243,23 @@ def get_resource_type(self, resource_type, region_name=None, config=None):
             resource_type, endpoint_url=endpoint_url, config=config, verify=self.verify
         )
 
+    @cached_property
+    def conn(self):
+        "Get the underlying boto3 client (cached)"
+        if self.client_type:
+            return self.get_client_type(self.client_type, region_name=self.region_name)
+        elif self.resource_type:
+            return self.get_resource_type(self.resource_type, region_name=self.region_name)
+        else:
+            raise AirflowException(
+                'Either self.client_type or self.resource_type'
+                ' must be specified in the subclass')
+
+    def get_conn(self):
+        "Get the underlying boto3 client (cached)"
```

Comment:
```suggestion
        """Get the underlying boto3 client (cached)"""
```

Reply:
Would be great if you could also add a type hint for the return value. Either in the description or the function header.

Reply:
Thanks, resolved